go/vt/vtgate/planbuilder/single_sharded_shortcut.go

```diff
@@ -44,47 +45,60 @@ func unshardedShortcut(ctx *plancontext.PlanningContext, stmt sqlparser.SelectSt
 
 	tableNames, err := getTableNames(ctx.SemTable)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	plan := &routeGen4{
 		eroute: &engine.Route{
 			RoutingParameters: &engine.RoutingParameters{
 				Opcode:   engine.Unsharded,
 				Keyspace: ks,
 			},
-			TableName: strings.Join(tableNames, ", "),
+			TableName: strings.Join(escapedTableNames(tableNames), ", "),
 		},
 		Select: stmt,
 	}
 
 	if err := plan.WireupGen4(ctx); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-	return plan, nil
+	return plan, operators.QualifiedTableNames(ks, tableNames), nil
 }
 
-func getTableNames(semTable *semantics.SemTable) ([]string, error) {
-	tableNameMap := map[string]any{}
+func escapedTableNames(tableNames []sqlparser.TableName) []string {
+	escaped := make([]string, len(tableNames))
+	for i, tableName := range tableNames {
+		escaped[i] = sqlparser.String(tableName)
+	}
+	return escaped
+}
+
+func getTableNames(semTable *semantics.SemTable) ([]sqlparser.TableName, error) {
+	tableNameMap := make(map[string]sqlparser.TableName)
 
 	for _, tableInfo := range semTable.Tables {
 		tblObj := tableInfo.GetVindexTable()
 		if tblObj == nil {
 			// probably a derived table
 			continue
 		}
-		var name string
 		if tableInfo.IsInfSchema() {
-			name = "tableName"
+			tableNameMap["tableName"] = sqlparser.TableName{
```

Comment:
Preserving logic that's already in the codebase, but don't understand what this is about.